registry/rpc/rpcregistry.go

```diff
@@ -183,6 +239,15 @@ func (r *RPCRegistry) SaveUnitState(unitName string, unitState *unit.UnitState,
 		defer debug.Exit_(debug.Enter_(unitName, unitState))
 	}
 
+	r.nocacheSaveUnitState(unitName, unitState, ttl)
+	r.cacheUnitState(unitName, unitState, ttl)
+}
+
+func (r *RPCRegistry) nocacheSaveUnitState(unitName string, unitState *unit.UnitState, ttl time.Duration) {
+	if DebugRPCRegistry {
+		defer debug.Exit_(debug.Enter_(unitName, unitState))
+	}
+
 	if unitState.UnitName == "" {
```

Comment:
I have to double check, but there is a case where the `unitState.UnitName == "" ` means something for fleet. It was used in the hearbeats but maybe somewhere else